Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -1151,7 +1151,7 @@ internal struct JSONScanner {
       throw JSONDecodingError.truncated
     }
     if currentByte != asciiDoubleQuote {
-      throw JSONDecodingError.malformedString
+      return nil
```

Comment:
It wasn't really an error before:  before this PR, the only caller of this method already checked this condition.

To clarify, I've just renamed this method to `nextOptionalKey()` to indicate that it returns `nil` if the text doesn't look like a quoted key.  It's now used as the fast path for both field name and enum parsing.  In both of those cases, we can use the `NameMap` to quickly resolve UTF8 bytes without converting to a String.